Userland/Libraries/LibVideo/VP9/Decoder.h

```diff
@@ -80,29 +131,87 @@ class Decoder {
+    u32 m_frame_height { 0 };
     u16 m_render_width { 0 };
     u16 m_render_height { 0 };
-    u16 m_mi_cols { 0 };
-    u16 m_mi_rows { 0 };
-    u16 m_sb64_cols { 0 };
-    u16 m_sb64_rows { 0 };
+    bool m_render_and_frame_size_different { false };
+    u32 m_mi_cols { 0 };
+    u32 m_mi_rows { 0 };
+    u32 m_sb64_cols { 0 };
+    u32 m_sb64_rows { 0 };
     InterpolationFilter m_interpolation_filter;
     bool m_lossless { false };
     u8 m_segmentation_tree_probs[7];
     u8 m_segmentation_pred_prob[3];
     bool m_feature_enabled[8][4];
     u8 m_feature_data[8][4];
+    bool m_segmentation_enabled { false };
+    bool m_segmentation_update_map { false };
+    bool m_segmentation_temporal_update { false };
     bool m_segmentation_abs_or_delta_update { false };
     u16 m_tile_cols_log2 { 0 };
     u16 m_tile_rows_log2 { 0 };
     i8 m_loop_filter_ref_deltas[MAX_REF_FRAMES];
     i8 m_loop_filter_mode_deltas[2];
 
+    u8** m_above_nonzero_context { nullptr };
+    u8** m_left_nonzero_context { nullptr };
+    u8* m_above_seg_pred_context { nullptr };
+    u8* m_left_seg_pred_context { nullptr };
+    u8* m_above_partition_context { nullptr };
+    u8* m_left_partition_context { nullptr };
+    u32 m_mi_row_start { 0 };
+    u32 m_mi_row_end { 0 };
+    u32 m_mi_col_start { 0 };
+    u32 m_mi_col_end { 0 };
+    u32 m_mi_row { 0 };
+    u32 m_mi_col { 0 };
+    u32 m_mi_size { 0 };
+    bool m_available_u { false };
+    bool m_available_l { false };
+    u8 m_segment_id { 0 };
+    bool m_skip { false };
+    u8 m_num_8x8 { 0 };
+    bool m_has_rows { false };
+    bool m_has_cols { false };
+    TXSize m_max_tx_size { TX_4x4 };
+    u8 m_block_subsize { 0 };
+    u32 m_row { 0 };
+    u32 m_col { 0 };
+    TXSize m_tx_size { TX_4x4 };
+    ReferenceFrame m_ref_frame[2];
+    bool m_is_inter { false };
+    IntraMode m_default_intra_mode { DcPred };
+    u8 m_y_mode { 0 };
+    u8 m_sub_modes[4]; // FIXME: What size is this supposed to be?
+    u8 m_num_4x4_w { 0 };
+    u8 m_num_4x4_h { 0 };
+    u8 m_uv_mode { 0 }; // FIXME: Is u8 the right size?
+    ReferenceFrame m_left_ref_frame[2];
+    ReferenceFrame m_above_ref_frame[2];
+    Vector<Vector<Vector<ReferenceFrame>>> m_ref_frames; // TODO: Can we make these fixed sized allocations?
+    bool m_left_intra { false };
+    bool m_above_intra { false };
+    bool m_left_single { false };
+    bool m_above_single { false };
+    Vector<Vector<u8>> m_prev_segment_ids;
+    InterpolationFilter m_interp_filter { EightTap };
+    InterMode m_mv[2];
+    InterMode m_near_mv[2];
+    InterMode m_nearest_mv[2];
+
+    bool m_use_hp { false };
+
+    TXMode m_tx_mode;
+    ReferenceMode m_reference_mode;
+    ReferenceFrame m_comp_fixed_ref;
+    ReferenceFrame m_comp_var_ref[2];
+
     OwnPtr<BitStream> m_bit_stream;
     OwnPtr<ProbabilityTables> m_probability_tables;
     OwnPtr<SyntaxElementCounter> m_syntax_element_counter;
+    OwnPtr<TreeParser> m_tree_parser;
```

Comment:
I think this could be a `NonnullOwnPtr` since it's assigned in the constructor and never cleared. Maybe it doesn't even need to be a `OwnPtr` and could be a simple `TreeParser`, depending on what you plan to do with it.